Sources/PayPalNativeCheckout/PayPalClient.swift

```diff
@@ -1,86 +1,78 @@
 import UIKit
-
-@_implementationOnly import PayPalCheckout
-
+import PayPalCheckout
 #if canImport(PaymentsCore)
 import PaymentsCore
 #endif
 
 /// PayPal Paysheet to handle PayPal transaction
+/// encapsulates instance to communicate with nxo
 public class PayPalClient {
 
     public weak var delegate: PayPalDelegate?
-
+    private let nativeCheckoutProvider: NativeCheckoutStartable
     private let apiClient: APIClient
     private let config: CoreConfig
 
-    // swiftlint:disable identifier_name
-    private let CheckoutFlow: CheckoutProtocol.Type
-    // swiftlint:enable identifier_name
-
     /// Initialize a PayPalClient to process PayPal transaction
     /// - Parameters:
     ///   - config: The CoreConfig object
     public convenience init(config: CoreConfig) {
         self.init(
             config: config,
-            checkoutFlow: Checkout.self,
+            nativeCheckoutProvider: NativeCheckoutProvider(),
             apiClient: APIClient(coreConfig: config)
         )
     }
 
-    init(config: CoreConfig, checkoutFlow: CheckoutProtocol.Type, apiClient: APIClient) {
+    init(config: CoreConfig, nativeCheckoutProvider: NativeCheckoutStartable, apiClient: APIClient) {
         self.config = config
-        self.CheckoutFlow = checkoutFlow
+        self.nativeCheckoutProvider = nativeCheckoutProvider
         self.apiClient = apiClient
     }
 
     /// Present PayPal Paysheet and start a PayPal transaction
     /// - Parameters:
-    ///   - request: the PayPalRequest for the transaction
     ///   - presentingViewController: the ViewController to present PayPalPaysheet on, if not provided, the Paysheet will be presented on your top-most ViewController
-    ///   - completion: Completion block to handle buyer's approval, cancellation, and error.
-    public func start(request: PayPalRequest, presentingViewController: UIViewController? = nil) async {
+    ///   - delegate: Completion block to handle buyer's approval, cancellation, error, create order callback, shipping change callback
```

Comment:
I think this is missing a docstring for the `orderID` param. I think we'd also want the orderID first since it's required?